libr/io/section.c

```diff
@@ -57,7 +57,10 @@ static RIOSection *findMatching (RIO *io, ut64 paddr, ut64 vaddr, ut64 size, ut6
 R_API RIOSection *r_io_section_add(RIO *io, ut64 offset, ut64 vaddr, ut64 size, ut64 vsize, int rwx, const char *name, ut32 bin_id, int fd) {
 	int update = 0;
 	RIOSection *s;
-	if (size == 0 || size > 0xf0000000) {
+	ut64 curr = lseek (fd, 0, SEEK_CUR);
```

Comment:
i think using r_io_desc_size is safer, much safer. r_io_size uses io->desc as RioDesc for r_io_desc_size. I'm not sure if the size+offset thing is needed